pages/create.tsx

```diff
@@ -54,6 +62,71 @@ const Home: NextPage = () => {
     }
   }
 
+  interface NengajoTokenMetadata {
+    name: string
+    image: string
+    description?: string | null | undefined
+    animation_url?: string | null | undefined
+    external_url?: string | null | undefined
+    attributes: TokenAttribute[]
+  }
+  interface TokenAttribute {
+    trait_type: string
+    value: number | string
+    display_type?: string | null | undefined
+    max_value?: number | null | undefined
+    trait_count?: number | null | undefined
+    order?: number | null | undefined
+  }
+  const metadata_description = 'A Nengajo sent using HENKAKU Nengajo.'
+  const metadata_external_url = 'https://henkaku-nengajo.vercel.app'
+
+  useEffect(() => {
```

Comment:
comment: 
There are pros and cons, There are also ways to avoid using "useEffect" so as to avoid its side effects.

ex: 
```tsx

  const handleSubmit = async () => {
    const imageUriPath = await sendFileToIPFS()
    if(imageUriPath) {
      setImageUri(imageUriPath)
      putMetadataOnIPFS(imageUriPath)
    }
  }
  const sendFileToIPFS = async () => {
    if (fileImg === undefined || fileImg === null) return false
    if (metadataName === '') return false
    setIsLoading(true)
    try {
      const formData = new FormData()
      formData.append('file', fileImg)
      const resFile = await axios({
        method: 'post',
        url: IPFS_API_ENDPOINT + '/pinning/pinFileToIPFS',
        data: formData,
        headers: {
          pinata_api_key: `${IPFS_API_KEY}`,
          pinata_secret_api_key: `${IPFS_API_SECRET}`,
          'Content-Type': 'multipart/form-data'
        }
      })
      const uri = `https://gateway.pinata.cloud/ipfs/${resFile.data.IpfsHash}`
      return uri
    } catch (error) {
      console.error('Error sending File to IPFS: ')
      console.error(error)
      return false
    }
  }

  const putMetadataOnIPFS = async (imageUriPath: string) => {
    if (metadataUri !== '') return
    if (address === undefined) return
    if (metadataName === '') return
    const metadata: NengajoTokenMetadata = {
      name: metadataName,
      description: metadata_description,
      image: imageUriPath,
      external_url: metadata_external_url,
      attributes: [
        {
          trait_type: 'Author',
          value: address,
          display_type: null,
          max_value: null,
          trait_count: 0,
          order: null
        }
      ]
    }
    console.log('Putting metadata on IPFS', metadata)
    const url = IPFS_API_ENDPOINT + '/pinning/pinJSONToIPFS'

    const pinataRequest = await fetch(url, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        pinata_api_key: `${IPFS_API_KEY}`,
        pinata_secret_api_key: `${IPFS_API_SECRET}`
      },
      body: JSON.stringify(metadata)
    })
    var res = await pinataRequest.json()
    setMetadataUri(`https://gateway.pinata.cloud/ipfs/${res.IpfsHash}`)
    console.log(
      'Metadata IPFS URI:',
      `https://gateway.pinata.cloud/ipfs/${res.IpfsHash}`
    )
  }
```

Reply:
Done on https://github.com/henkaku-center/henkaku-nengajo-frontend/pull/37/commits/3d25f7d413227d264506d047ccd6c0d10bc97c90

Reply:
Please see my comment here.
https://github.com/henkaku-center/henkaku-nengajo-frontend/commit/3d25f7d413227d264506d047ccd6c0d10bc97c90#r92869373

Reply:
Ooooh! I see. Sorry, I thought I understood what you did and they did it myself, but I missed this part. I see you already have "return false" on your original comment.

I'm doubly sorry because you took the trouble to share that code, and then took the trouble to check my commit, and found that I didn't spend as long with your code.

I will check thoroughly and push a commit 🙇

Reply:
No problem!
I'm sorry it was so hard to understand🙇

Reply:
Done on https://github.com/henkaku-center/henkaku-nengajo-frontend/pull/37/commits/94c8acb7c9ea61278938108f4fc5349c34ef2d2b

It wasn't hard to understand, I just didn't pay enough attention 🥲

Reply:
LGTM!

Reply:
Thank you! Could you approve your review on this PR? (The `main` branch is not protected, but I guess it's better form if someone approves the PRs before merging.)

I will add the couple of issues that the PR and the conversation introduced.